Composer/packages/server/src/controllers/project.ts

```diff
@@ -236,8 +242,15 @@ async function updateLgFile(req: Request, res: Response) {
 
   const currentProject = await BotProjectService.getProjectById(projectId, user);
   if (currentProject !== undefined) {
-    const lgFiles = await currentProject.updateLgFile(req.body.id, req.body.content);
-    res.status(200).json({ lgFiles });
+    const file = currentProject.lgFiles.find(d => d.id === req.body.id);
+    if (file && new Date(file.lastModified).getTime() > new Date(req.body.lastModified).getTime()) {
+      res.status(409).json({ message: 'This content was updated since it was loaded. Please reload.' });
```

Comment:
We need to allow translation for this string. Typically we can send a status code to the client and have the client own the copy.